File: Backend/Microservices/UsersService/UsersService.Domain/Entities/TrainerWorkoutProgram.cs
﻿using Common.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UsersService.Domain.Entities;

public class TrainerWorkoutProgram : AuditableBaseEntity
{
  public string TrainerSubjectId { get; set; } = default!;
  public string Name { get; set; } = default!;
  public string Title { get; set; } = default!;
  public string Description { get; set; } = default!;
  public string HeaderImageUrl { get; set; } = default!;
  public string ProgramDetails { get; set; } = default!;
  public double Price { get; set; } = default!;
  public virtual List<User> EnrolledUsers { get; set; } = new List<User>();
}

File: MobileApp/gymly/lib/models/trainer_workout_program.dart
class TrainerWorkoutProgram {
  final int id;
  final String name;
  final String title;
  final String description;
  final String headerImageUrl;
  final String programDetails;
  final double price;

  TrainerWorkoutProgram(this.id, this.name, this.title, this.description,
      this.headerImageUrl, this.programDetails, this.price);

  factory TrainerWorkoutProgram.fromJson(Map<String, dynamic> json) {
    return TrainerWorkoutProgram(
      json["id"] as int,
      json["name"] as String,
      json["title"] as String,
      json["description"] as String,
      json["headerImageUrl"] as String,
      json["programDetails"] as String,
      double.tryParse(json["price"].toString())!,
    );
  }
}
